python/examples/advanced_agents/indie_hacker_agent/langgraph/main.py

```diff
@@ -0,0 +1,75 @@
+from composio import Composio, ComposioToolSet
+import os
+from dotenv import load_dotenv
+from langchain_groq import ChatGroq
+from langchain_openai import ChatOpenAI
+from langgraph.checkpoint.memory import MemorySaver
+from typing import Annotated
+from langchain_core.messages import BaseMessage
+from typing_extensions import TypedDict
+from composio_langgraph import Action, ComposioToolSet, App
+from langgraph.graph import StateGraph, START, END
+from langgraph.graph.message import add_messages
+from langgraph.prebuilt import ToolNode, tools_condition
+load_dotenv()
+
+class State(TypedDict):
+    messages: Annotated[list, add_messages]
+
+graph_builder = StateGraph(State)
+
+toolset = ComposioToolSet()
+tools = toolset.get_tools(apps=[App.CODEINTERPRETER, App.FILETOOL, App.HACKERNEWS, App.SHELLTOOL, App.WEBTOOL])
+
+llm = ChatOpenAI(model="gpt-4o")
```

Comment:
The model configuration is hardcoded and commented out. Consider using environment variables to switch between models (OpenAI vs Groq) for better flexibility.